RESTserver/src/main/java/com/softserve/if072/restservice/controller/HistoryController.java

```diff
@@ -83,16 +81,11 @@ public void deleteAll(@PathVariable int userId) {
         historyService.deleteAll(userId);
     }
 
-    @PostAuthorize("#history != null && #history.user != null"
-            + " && #history.user.id == authentication.user.id")
+    @PreAuthorize("@historySecurityService.hasPermissionToAccessByProductId(#productId)")
     @GetMapping("/products/{productId}")
     @ResponseStatus(HttpStatus.OK)
     public List<History> getByProductId(@PathVariable int userId, @PathVariable int productId) {
         List<History> histories = historyService.getByProductId(userId, productId);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "histories". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1488)